File: programme/pattern-3/pattern-3.py
# Write a Python programme to print the given pattern

def print_pattern(n: int) -> None:
    print("Output : \n")
    for col in range(n):
        for row in range(n):
            if col == 0 or col == n - 1 or row == 0 or row == n - 1:
                print(' * ', end='')
            else:
                print('   ', end='')
        print()


n = int(input("input  : "))
print_pattern(n)

File: programme/pattern-3/pattern-3.js
// Write a python programme to print the given pattern

"use strict";

// perform following command to run programme
// node pattern-3.js

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

function pattern(n) {
  for (let i = 0; i < n - 1; i++) {
    rl.write("x ");
  }
  rl.write("x\n");
  for (let i = 0; i < n - 2; i++) {
    rl.write("x");
    for (let i = 0; i < n - 2; i++) {
      rl.write("  ");
    }
    rl.write(" x\n");
  }
  if (n != 1) {
    for (let i = 0; i < n - 1; i++) {
      rl.write("x ");
    }
    rl.write("x\n");
  }
}

rl.question("Input  : ", (n) => {
  console.log("Output :\n");
  n = parseInt(n);
  if (isNaN(n)) {
    console.error("Some error occured");
  } else {
    pattern(n);
  }
  rl.close();
});
